maven-plugin/src/main/java/org/openmrs/maven/plugins/Clone.java

```diff
@@ -0,0 +1,206 @@
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.StoredConfig;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.openmrs.maven.plugins.utility.SDKConstants;
+import org.twdata.maven.mojoexecutor.MojoExecutor;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.twdata.maven.mojoexecutor.MojoExecutor.*;
+
+/**
+ *  @goal clone
+ *
+ */
+
+public class Clone extends AbstractTask {
+
+    /**
+     * @parameter expression="${groupId}" default-value="org.openmrs.module"
+     */
+    private String groupId;
+
+    /**
+     * @parameter expression="${artifactId}"
+     */
+    private String artifactId;
+
+    /**
+     * @parameter expression="${githubUsername}"
+     */
+    private String githubUsername;
+
+    /**
+     * @parameter expression="${githubPassword}"
+     */
+    private String githubPassword;
+
+    private static final String GOAL_COPY = "copy";
+    private static final String OPENMRS_GITHUB_KEY_PARTIAL = "/openmrs/";
+
+    @Override
+    public void executeTask() throws MojoExecutionException, MojoFailureException {
+
+        if (artifactId == null) {
+            artifactId = wizard.promptForValueIfMissing(null, "artifactId");
+        }
+
+        String version = versionsHelper.getLatestReleasedVersion(new Artifact(artifactId, "0", groupId));
+        String gitHubHttpKey;
+
+        try {
+            gitHubHttpKey = extractGitHubHttpKeyFromModulePom(artifactId, version, groupId);
+        } catch (IOException e) {
+            throw new IllegalStateException("Failed to fetch github data from module repository");
+        }
+
+        if (githubUsername == null) {
+            githubUsername = wizard.promptForValueIfMissing(null, "GitHub username");
+            // Remove eventually mistakes like spaces, tabs etc.
+            githubUsername = githubUsername.replaceAll("\\s+", "");
+        }
+
+        if (githubPassword == null) {
+            githubPassword = wizard.promptForValueIfMissing(null, "GitHub password");
+        }
+
+        cloneRepo(gitHubHttpKey);
+
+        System.out.println("");
+    }
+
+    private String extractGitHubHttpKeyFromModulePom(String artifactId, String version, String groupId) throws MojoExecutionException, IOException {
+        downloadModulePom(new Artifact(artifactId, version, groupId, "pom"));
+
+        File pomDir = new File("pom/");
+        File pom = new File(pomDir ,this.artifactId+"-"+version+".pom");
+
+        BufferedReader br = null;
+        br = new BufferedReader(new FileReader(pom));
+        String url = "";
+        String line = br.readLine();
+        while (line != null) {
+            if (line.contains("https://github.com/")) {
+                url = line.substring(line.indexOf(">")+1, line.indexOf("</"));
+                break;
```

Comment:
That one was tricky since Project.hasProject() method inside Project.loadProject() method.
hasProject() were checking if there is file names exactly "pom.xml", but pom files downloaded by maven has different naming convention (for example webservices.rest's pom is called "webservices.rest-2.14.pom")
I've modified and overloaded loadProject() method, so now its possible to pass custom filename.
Fixed